core/tests/protractor_desktop/contributorDashboard.js

```diff
@@ -16,65 +16,318 @@
  * @fileoverview End-to-end tests for the contributor dashboard page.
  */
 
+var forms = require('../protractor_utils/forms.js');
 var general = require('../protractor_utils/general.js');
 var users = require('../protractor_utils/users.js');
+var workflow = require('../protractor_utils/workflow.js');
 var waitFor = require('../protractor_utils/waitFor.js');
 
 var AdminPage = require('../protractor_utils/AdminPage.js');
 var ContributorDashboardPage = require(
   '../protractor_utils/ContributorDashboardPage.js');
+var ExplorationEditorPage = require(
+  '../protractor_utils/ExplorationEditorPage.js');
+var SkillEditorPage = require(
+  '../protractor_utils/SkillEditorPage.js');
+var TopicsAndSkillsDashboardPage = require(
+  '../protractor_utils/TopicsAndSkillsDashboardPage.js');
 
-describe('Contributor Dashboard page', function() {
-  var contributorDashboardPage = null;
-  var contributorDashboardTranslateTextTab = null;
+describe('Contributor dashboard page', function() {
+  const TOPIC_NAMES = [
+    'Topic 0 for contribution', 'Topic 1 for contribution'];
+  const SKILL_DESCRIPTIONS = [
+    'Skill 0 for suggestion', 'Skill 1 for suggestion'];
+  const REVIEW_MATERIALS = [
+    'Review Material 0',
+    'Review Material 1'];
+  const ADMIN_EMAIL = 'management@contributor.com';
+  const USER_EMAILS = ['user0@contributor.com', 'user1@contributor.com'];
+  const HINDI_LANGUAGE = 'Hindi';
+  let contributorDashboardPage = null;
+  let contributorDashboardTranslateTextTab = null;
+  let topicsAndSkillsDashboardPage = null;
+  let skillEditorPage = null;
+  let explorationEditorPage = null;
+  let explorationEditorMainTab = null;
+  let adminPage = null;
 
   beforeAll(async function() {
     contributorDashboardPage = (
       new ContributorDashboardPage.ContributorDashboardPage());
     contributorDashboardTranslateTextTab = (
       contributorDashboardPage.getTranslateTextTab());
+    topicsAndSkillsDashboardPage =
+      new TopicsAndSkillsDashboardPage.TopicsAndSkillsDashboardPage();
+    skillEditorPage =
+      new SkillEditorPage.SkillEditorPage();
+    explorationEditorPage = new ExplorationEditorPage.ExplorationEditorPage();
+    explorationEditorMainTab = explorationEditorPage.getMainTab();
+    adminPage = new AdminPage.AdminPage();
+    await users.createUser(USER_EMAILS[0], 'user0');
+    await users.createUser(USER_EMAILS[1], 'user1');
+    await users.createAndLoginAdminUser(ADMIN_EMAIL, 'management');
+    // Create 2 topics and 2 skills. Link 1 skill to 1 topic.
+    await topicsAndSkillsDashboardPage.get();
+    await topicsAndSkillsDashboardPage.createTopic(
+      TOPIC_NAMES[0], 'abbrev-one', 'Topic description 1', false);
```

Comment:
Here and below, can you make the url_fragment field name (2nd argument) unique to this test? It is expected to be a globally unique field. This is one of the review checks mentioned [here](https://github.com/oppia/oppia/wiki/End-to-End-Tests#codeowner-checks).

Reply:
Sorry, I didn't get it by unique you, should I change both of this such that we there's less chance of someone using the same name in another suite? Can you give some examples?

I'm following this pattern:

![image](https://user-images.githubusercontent.com/16653571/89966121-0cf23500-dc6c-11ea-9533-2ad4380e42f3.png)

Reply:
Yep. That's exactly what I meant. You can could use the test name (or an abbreviated version of the test name), to make the value unique across the server. See example below:
https://github.com/oppia/oppia/blob/5c8475f8231473a931110bc4133f9110e2f7de56/core/tests/protractor_desktop/topicsAndSkillsDashboard.js#L58-L59

Reply:
Changed it to `topic-for-community-one`. Done!